datarepo-clienttests/src/main/java/scripts/testscripts/SnapshotScaleDelete.java

```diff
@@ -0,0 +1,124 @@
+
+  private int NUM_SNAPSHOTS = 1;
+
+  public void setParameters(List<String> parameters) {
+    if (parameters != null && parameters.size() > 0) {
+      NUM_SNAPSHOTS = Integer.parseInt(parameters.get(0));
+    }
+    logger.debug("Number of snapshots to create and delete (default is 1): {}", NUM_SNAPSHOTS);
+  }
+
+  public void setup(List<TestUserSpecification> testUsers) throws Exception {
+    // create the profile and dataset
+    super.setup(testUsers);
+
+    // get the ApiClient for the snapshot creator, same as the dataset creator
+    ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
+    RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+
+    // dont bother to load data into the new dataset
+    // create MANY snapshots based on a single dataset
+
+    /*    for (int i = 0; i < NUM_SNAPSHOTS; i++) {
+       // make the create snapshot request and wait for the job to finish
+       // the name of the snapshot will already be randomized based on the bool
+       JobModel createSnapshotJobResponse =
+           DataRepoUtils.createSnapshot(
+               repositoryApi, datasetSummaryModel, "snapshot-simple.json", true);
+
+       // save a reference to the snapshot summary model so we can delete it in cleanup()
+       // TODO ^ shouldn't I be deleting them all anyway?
+       // do I need to account for the fact that many will be deleted already?
+       // another idea would be that we create and then delete them each in the userjourney
+       SnapshotSummaryModel snapshotSummaryModel =
+           DataRepoUtils.expectJobSuccess(
+               repositoryApi, createSnapshotJobResponse, SnapshotSummaryModel.class);
+       logger.info("Successfully created snapshot: {}, index: {}",
+           snapshotSummaryModel.getName(),
+           i
+       );
+    }*/
+  }
+
+  public void userJourney(TestUserSpecification testUser) throws Exception {
+    ApiClient apiClient = DataRepoUtils.getClientForTestUser(testUser, server);
+    DataRepositoryServiceApi dataRepositoryServiceApi = new DataRepositoryServiceApi(apiClient);
+
+    for (int i = 0; i < NUM_SNAPSHOTS; i++) {
+      // make the create snapshot request and wait for the job to finish
+      // the name of the snapshot will already be randomized based on the bool
+      JobModel createSnapshotJobResponse =
+          DataRepoUtils.createSnapshot(
+              repositoryApi, datasetSummaryModel, "snapshot-simple.json", true);
+
+      // save a reference to the snapshot summary model so we can delete it in cleanup(
+      SnapshotSummaryModel snapshotSummaryModel =
+          DataRepoUtils.expectJobSuccess(
+              repositoryApi, createSnapshotJobResponse, SnapshotSummaryModel.class);
+      logger.info(
+          "Successfully created snapshot: {}, index: {}", snapshotSummaryModel.getName(), i);
+
+      // Now delete each snapshot
+      JobModel deleteSnapshotJobResponse =
```

Comment:
If this test fails before it reaches this point, the snapshots won't be deleted. Is this OK?

Reply:
If the test throws an exception, it calls the cleanup method which will delete any snapshots that were created by the test.

Reply:
So it sounds like we don't have another cleanup mechanism other than the cleanup method, so this is the best we can do here.